Build failure: mysql84 and mysql80 on ARM MacOS

## Steps To Reproduce

Steps to reproduce the behavior:

Attempt to build pkgs.mysql80 or pkgs.mysql84 on an M1 Mac, for example:
`nix-shell -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/master.tar.gz -p pkgs.mysql84`

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/d90wypg24ikik6vw9hlgwkz3jl26j1zk-mysql-8.4.3.tar.gz
source root is mysql-8.4.3
setting SOURCE_DATE_EPOCH to timestamp 1726571427 of file mysql-8.4.3/sql/sql_yacc.h
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/43mi2gqgsvxsvvjdyfggwk6wa9sk0fkj-no-force-outline-atomics.patch
patching file CMakeLists.txt
Hunk #1 succeeded at 1344 (offset 6 lines).
substituteStream() in derivation mysql-8.4.3: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'cmake/libutils.cmake')
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./extra/icu/icu-release-73-1/source/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./extra/tirpc/libtirpc-1.3.5/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./extra/icu/icu-release-73-1/source/config.guess
Updating Autotools / GNU config script to a newer upstream version: ./extra/tirpc/libtirpc-1.3.5/config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/ilbxhxgy6d72p688cfcsrzj02b5w32jy-mysql-8.4.3/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/ilbxhxgy6d72p688cfcsrzj02b5w32jy-mysql-8.4.3/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/ilbxhxgy6d72p688cfcsrzj02b5w32jy-mysql-8.4.3/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/ilbxhxgy6d72p688cfcsrzj02b5w32jy-mysql-8.4.3/share/doc/mysql -DCMAKE_INSTALL_INFODIR=/nix/store/ilbxhxgy6d72p688cfcsrzj02b5w32jy-mysql-8.4.3/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/ilbxhxgy6d72p688cfcsrzj02b5w32jy-mysql-8.4.3/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/ilbxhxgy6d72p688cfcsrzj02b5w32jy-mysql-8.4.3/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/ilbxhxgy6d72p688cfcsrzj02b5w32jy-mysql-8.4.3/include -DCMAKE_INSTALL_SBINDIR=/nix/store/ilbxhxgy6d72p688cfcsrzj02b5w32jy-mysql-8.4.3/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/ilbxhxgy6d72p688cfcsrzj02b5w32jy-mysql-8.4.3/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/ilbxhxgy6d72p688cfcsrzj02b5w32jy-mysql-8.4.3/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/wkadfwbhn9b4kdih88j7izzy20qw79bk-clang-wrapper-19.1.5/bin/strip -DCMAKE_RANLIB=/nix/store/wkadfwbhn9b4kdih88j7izzy20qw79bk-clang-wrapper-19.1.5/bin/ranlib -DCMAKE_AR=/nix/store/wkadfwbhn9b4kdih88j7izzy20qw79bk-clang-wrapper-19.1.5/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/ilbxhxgy6d72p688cfcsrzj02b5w32jy-mysql-8.4.3 -DFORCE_UNSUPPORTED_COMPILER=1 -DWITH_ROUTER=OFF -DWITH_SYSTEM_LIBS=ON -DWITH_UNIT_TESTS=OFF -DMYSQL_UNIX_ADDR=/run/mysqld/mysqld.sock -DMYSQL_DATADIR=/var/lib/mysql -DINSTALL_INFODIR=share/mysql/docs -DINSTALL_MANDIR=share/man -DINSTALL_PLUGINDIR=lib/mysql/plugin -DINSTALL_INCLUDEDIR=include/mysql -DINSTALL_DOCREADMEDIR=share/mysql -DINSTALL_SUPPORTFILESDIR=share/mysql -DINSTALL_MYSQLSHAREDIR=share/mysql -DINSTALL_MYSQLTESTDIR= -DINSTALL_DOCDIR=share/mysql/docs -DINSTALL_SHAREDIR=share/mysql
-- Running cmake version 3.30.5
-- Could NOT find Git (missing: GIT_EXECUTABLE) 
Hint: The project() command has not yet been called.  It sets up system-specific search paths.
-- CMAKE_MODULE_PATH is /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/cmake
-- MySQL 8.4.3
-- The C compiler identification is Clang 19.1.5
-- The CXX compiler identification is Clang 19.1.5
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/wkadfwbhn9b4kdih88j7izzy20qw79bk-clang-wrapper-19.1.5/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/wkadfwbhn9b4kdih88j7izzy20qw79bk-clang-wrapper-19.1.5/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Source directory /private/tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3
-- Binary directory /private/tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/build
-- CMAKE_GENERATOR: Unix Makefiles
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of void *
-- Check size of void * - done
-- SIZEOF_VOIDP 8
-- DEFAULT_PLATFORM macos15.2
-- Packaging as: mysql-8.4.3-macos15.2-arm64
-- BOOST_PATCHES_DIR /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/include/boost_1_84_0/patches
-- BOOST_INCLUDE_DIR /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/extra/boost/boost_1_84_0
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Looking for floor
-- Looking for floor - found
-- Looking for log
-- Looking for log - found
-- Looking for gethostbyname_r
-- Looking for gethostbyname_r - not found
-- Looking for gethostbyname_r in nsl_r
-- Looking for gethostbyname_r in nsl_r - not found
-- Looking for gethostbyname_r in nsl
-- Looking for gethostbyname_r in nsl - not found
-- Looking for bind
-- Looking for bind - found
-- Looking for setsockopt
-- Looking for setsockopt - found
-- Looking for dlopen
-- Looking for dlopen - found
-- Looking for dlsym
-- Looking for dlsym - found
-- Looking for sched_yield
-- Looking for sched_yield - found
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Looking for timer_create
-- Looking for timer_create - not found
-- Looking for timer_create in rt
-- Looking for timer_create in rt - not found
-- Looking for backtrace
-- Looking for backtrace - found
-- Looking for include file alloca.h
-- Looking for include file alloca.h - found
-- Looking for include file arpa/inet.h
-- Looking for include file arpa/inet.h - found
-- Looking for include file dlfcn.h
-- Looking for include file dlfcn.h - found
-- Looking for include file endian.h
-- Looking for include file endian.h - not found
-- Looking for include file execinfo.h
-- Looking for include file execinfo.h - found
-- Looking for include file fpu_control.h
-- Looking for include file fpu_control.h - not found
-- Looking for include file grp.h
-- Looking for include file grp.h - found
-- Looking for include file langinfo.h
-- Looking for include file langinfo.h - found
-- Looking for include file malloc.h
-- Looking for include file malloc.h - not found
-- Looking for include file netinet/in.h
-- Looking for include file netinet/in.h - found
-- Looking for include file poll.h
-- Looking for include file poll.h - found
-- Looking for include file pwd.h
-- Looking for include file pwd.h - found
-- Looking for include file strings.h
-- Looking for include file strings.h - found
-- Looking for include file sys/ioctl.h
-- Looking for include file sys/ioctl.h - found
-- Looking for include file sys/mman.h
-- Looking for include file sys/mman.h - found
-- Looking for include file sys/prctl.h
-- Looking for include file sys/prctl.h - not found
-- Looking for include file sys/resource.h
-- Looking for include file sys/resource.h - found
-- Looking for include file sys/select.h
-- Looking for include file sys/select.h - found
-- Looking for include file sys/socket.h
-- Looking for include file sys/socket.h - found
-- Looking for include files curses.h, term.h
-- Looking for include files curses.h, term.h - found
-- Looking for include file termios.h
-- Looking for include file termios.h - found
-- Looking for include file termio.h
-- Looking for include file termio.h - not found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found
-- Looking for include file sys/wait.h
-- Looking for include file sys/wait.h - found
-- Looking for include file sys/param.h
-- Looking for include file sys/param.h - found
-- Looking for include file fnmatch.h
-- Looking for include file fnmatch.h - found
-- Looking for include file sys/un.h
-- Looking for include file sys/un.h - found
-- Looking for include files sys/types.h, sasl/sasl.h
-- Looking for include files sys/types.h, sasl/sasl.h - not found
-- Looking for _aligned_malloc
-- Looking for _aligned_malloc - not found
-- Looking for backtrace
-- Looking for backtrace - found
-- Looking for index
-- Looking for index - found
-- Looking for chown
-- Looking for chown - found
-- Looking for cuserid
-- Looking for cuserid - not found
-- Looking for directio
-- Looking for directio - not found
-- Looking for ftruncate
-- Looking for ftruncate - found
-- Looking for fchmod
-- Looking for fchmod - found
-- Looking for fcntl
-- Looking for fcntl - found
-- Looking for fdatasync
-- Looking for fdatasync - found
-- Looking for fdatasync
-- Looking for fdatasync - not found
-- Looking for fedisableexcept
-- Looking for fedisableexcept - not found
-- Looking for fsync
-- Looking for fsync - found
-- Looking for gethrtime
-- Looking for gethrtime - not found
-- Looking for getpass
-- Looking for getpass - found
-- Looking for getpassphrase
-- Looking for getpassphrase - not found
-- Looking for getpwnam
-- Looking for getpwnam - found
-- Looking for getpwuid
-- Looking for getpwuid - found
-- Looking for getrusage
-- Looking for getrusage - found
-- Looking for initgroups
-- Looking for initgroups - found
-- Looking for issetugid
-- Looking for issetugid - found
-- Looking for getuid
-- Looking for getuid - found
-- Looking for geteuid
-- Looking for geteuid - found
-- Looking for getgid
-- Looking for getgid - found
-- Looking for getegid
-- Looking for getegid - found
-- Looking for madvise
-- Looking for madvise - found
-- Looking for malloc_info
-- Looking for malloc_info - not found
-- Looking for mlock
-- Looking for mlock - found
-- Looking for mlockall
-- Looking for mlockall - found
-- Looking for mmap64
-- Looking for mmap64 - not found
-- Looking for poll
-- Looking for poll - found
-- Looking for posix_fallocate
-- Looking for posix_fallocate - not found
-- Looking for posix_memalign
-- Looking for posix_memalign - found
-- Looking for pthread_condattr_setclock
-- Looking for pthread_condattr_setclock - not found
-- Looking for pthread_getaffinity_np
-- Looking for pthread_getaffinity_np - not found
-- Looking for pthread_sigmask
-- Looking for pthread_sigmask - found
-- Looking for sleep
-- Looking for sleep - found
-- Looking for stpcpy
-- Looking for stpcpy - found
-- Looking for stpncpy
-- Looking for stpncpy - found
-- Looking for strlcpy
-- Looking for strlcpy - found
-- Looking for strndup
-- Looking for strndup - found
-- Looking for strlcat
-- Looking for strlcat - found
-- Looking for strptime
-- Looking for strptime - found
-- Looking for strsignal
-- Looking for strsignal - found
-- Looking for tell
-- Looking for tell - not found
-- Looking for vasprintf
-- Looking for vasprintf - found
-- Looking for memalign
-- Looking for memalign - not found
-- Looking for nl_langinfo
-- Looking for nl_langinfo - found
-- Looking for ntohll
-- Looking for ntohll - not found
-- Looking for epoll_create
-- Looking for epoll_create - not found
-- Looking for include file sys/time.h
-- Looking for include file sys/time.h - found
-- Looking for include file sys/times.h
-- Looking for include file sys/times.h - found
-- Looking for times
-- Looking for times - found
-- Looking for gettimeofday
-- Looking for gettimeofday - found
-- Looking for lrand48
-- Looking for lrand48 - found
-- Looking for TIOCGWINSZ
-- Looking for TIOCGWINSZ - found
-- Looking for FIONREAD
-- Looking for FIONREAD - found
-- Looking for FIONREAD
-- Looking for FIONREAD - found
-- Looking for MADV_DONTDUMP
-- Looking for MADV_DONTDUMP - not found
-- Performing Test HAVE_O_TMPFILE
-- Performing Test HAVE_O_TMPFILE - Failed
-- Looking for timer_create
-- Looking for timer_create - not found
-- Looking for timer_settime
-- Looking for timer_settime - not found
-- Looking for kqueue
-- Looking for kqueue - found
-- Performing Test HAVE_SETNS
-- Performing Test HAVE_SETNS - Failed
-- Looking for EVFILT_TIMER
-- Looking for EVFILT_TIMER - found
-- Check size of char *
-- Check size of char * - done
-- Check size of long
-- Check size of long - done
-- Check size of short
-- Check size of short - done
-- Check size of int
-- Check size of int - done
-- Check size of long long
-- Check size of long long - done
-- Check size of time_t
-- Check size of time_t - done
-- Performing Test HAVE_TM_GMTOFF
-- Performing Test HAVE_TM_GMTOFF - Success
-- Check size of ulong
-- Check size of ulong - failed
-- Check size of u_int32_t
-- Check size of u_int32_t - done
-- Performing Test HAVE_VISIBILITY_HIDDEN
-- Performing Test HAVE_VISIBILITY_HIDDEN - Success
-- Performing Test HAVE_CLOCK_GETTIME
-- Performing Test HAVE_CLOCK_GETTIME - Success
-- Performing Test HAVE_CLOCK_REALTIME
-- Performing Test HAVE_CLOCK_REALTIME - Success
-- Looking for include files time.h, sys/time.h
-- Looking for include files time.h, sys/time.h - found
-- Looking for O_NONBLOCK
-- Looking for O_NONBLOCK - found
-- Performing Test HAVE_FAKE_PAUSE_INSTRUCTION
-- Performing Test HAVE_FAKE_PAUSE_INSTRUCTION - Failed
-- Performing Test HAVE_HMT_PRIORITY_INSTRUCTION
-- Performing Test HAVE_HMT_PRIORITY_INSTRUCTION - Failed
-- Looking for C++ include cxxabi.h
-- Looking for C++ include cxxabi.h - found
-- Performing Test HAVE_ABI_CXA_DEMANGLE
-- Performing Test HAVE_ABI_CXA_DEMANGLE - Success
-- Performing Test HAVE_BUILTIN_UNREACHABLE
-- Performing Test HAVE_BUILTIN_UNREACHABLE - Success
-- Performing Test HAVE_BUILTIN_EXPECT
-- Performing Test HAVE_BUILTIN_EXPECT - Success
-- Looking for __builtin_stpcpy
-- Looking for __builtin_stpcpy - not found
-- Performing Test HAVE_GCC_SYNC_BUILTINS
-- Performing Test HAVE_GCC_SYNC_BUILTINS - Success
-- Performing Test HAVE_SYS_GETTID
-- Performing Test HAVE_SYS_GETTID - Success
-- Performing Test HAVE_PTHREAD_GETTHREADID_NP
-- Performing Test HAVE_PTHREAD_GETTHREADID_NP - Failed
-- Performing Test HAVE_PTHREAD_THREADID_NP
-- Performing Test HAVE_PTHREAD_THREADID_NP - Success
-- Performing Test HAVE_INTEGER_PTHREAD_SELF
-- Performing Test HAVE_INTEGER_PTHREAD_SELF - Failed
-- Performing Test HAVE_PTHREAD_SETNAME_NP_LINUX
-- Performing Test HAVE_PTHREAD_SETNAME_NP_LINUX - Failed
-- Performing Test HAVE_PTHREAD_SETNAME_NP_MACOS
-- Performing Test HAVE_PTHREAD_SETNAME_NP_MACOS - Success
-- Looking for netinet/in6.h
-- Looking for netinet/in6.h - not found
-- Performing Test HAVE_SOCKADDR_IN_SIN_LEN
-- Performing Test HAVE_SOCKADDR_IN_SIN_LEN - Success
-- Performing Test HAVE_SOCKADDR_IN6_SIN6_LEN
-- Performing Test HAVE_SOCKADDR_IN6_SIN6_LEN - Success
-- Looking for include file numa.h
-- Looking for include file numa.h - not found
-- Looking for include file numaif.h
-- Looking for include file numaif.h - not found
-- NUMA library missing or required version not available
-- Adding -DWITH_CURL=system
-- Adding -DWITH_EDITLINE=system
-- Adding -DWITH_ICU=system
-- Adding -DWITH_LZ4=system
-- Adding -DWITH_PROTOBUF=system
-- Adding -DWITH_SSL=system
-- Adding -DWITH_ZSTD=system
-- Performing Test HAVE_CXX_W_array_bounds
-- Performing Test HAVE_CXX_W_array_bounds - Success
-- ZLIB_VERSION (bundled) is 1.3.1
-- ZLIB_INCLUDE_DIR /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/extra/zlib/zlib-1.3.1
-- Performing Test HAVE_CXX_W_missing_profile
-- Performing Test HAVE_CXX_W_missing_profile - Failed
-- Performing Test HAS_DEPRECATED_NON_PROTOTYPE
-- Performing Test HAS_DEPRECATED_NON_PROTOTYPE - Success
-- Check size of off64_t
-- Check size of off64_t - failed
-- Looking for fseeko
-- Looking for fseeko - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- ZSTD_VERSION (system) is 1.5.6
-- ZSTD_INCLUDE_DIR /nix/store/jsl7ybc00wl8nhqxirq3iwhckpj6c18c-zstd-1.5.6-dev/include
-- Found OpenSSL: /nix/store/8ms7abrwg2nbqdjdcdf1aig4y7yha4v7-openssl-3.3.2/lib/libcrypto.dylib (found version "3.3.2")
-- OPENSSL_VERSION (system) is 3.3.2
-- OPENSSL_INCLUDE_DIR = /nix/store/l2nvwi0k7nk9azm42yr6zkf2b2bmf0y7-openssl-3.3.2-dev/include
-- OPENSSL_LIBRARIES = /nix/store/8ms7abrwg2nbqdjdcdf1aig4y7yha4v7-openssl-3.3.2/lib/libssl.dylib /nix/store/8ms7abrwg2nbqdjdcdf1aig4y7yha4v7-openssl-3.3.2/lib/libcrypto.dylib
-- KERBEROS path is none, disabling kerberos support.
-- HAVE_KRB5_KRB5_H  
-- KERBEROS_LIBRARIES  
CMake Warning at cmake/sasl.cmake:280 (MESSAGE):
  Could not find SASL
Call Stack (most recent call first):
  CMakeLists.txt:1826 (MYSQL_CHECK_SASL)


-- Looking for lber.h
-- Looking for lber.h - not found
-- Looking for ldap.h
-- Looking for ldap.h - not found
CMake Warning at cmake/ldap.cmake:159 (MESSAGE):
  Could not find LDAP
Call Stack (most recent call first):
  CMakeLists.txt:1830 (MYSQL_CHECK_LDAP)


-- Check size of mbstate_t
-- Check size of mbstate_t - done
-- Performing Test HAVE_LANGINFO_CODESET
-- Performing Test HAVE_LANGINFO_CODESET - Success
-- Looking for wcsdup
-- Looking for wcsdup - found
-- Check size of wchar_t
-- Check size of wchar_t - done
-- Check size of wint_t
-- Check size of wint_t - done
-- EDITLINE_INCLUDE_DIR /nix/store/6ijk0dfsvcrbd2spxsprc8n0y2fmiwsq-libedit-20240808-3.1-dev/include/editline
-- EDITLINE_LIBRARY /nix/store/8x1kqz8573gsz9m2h7mslrif44hybwmv-libedit-20240808-3.1/lib/libedit.dylib
-- Performing Test EDITLINE_HAVE_HIST_ENTRY
-- Performing Test EDITLINE_HAVE_HIST_ENTRY - Success
-- Performing Test EDITLINE_HAVE_COMPLETION_INT
-- Performing Test EDITLINE_HAVE_COMPLETION_INT - Failed
-- Performing Test EDITLINE_HAVE_COMPLETION_CHAR
-- Performing Test EDITLINE_HAVE_COMPLETION_CHAR - Success
-- LZ4_VERSION (system) is 1.10.0
-- LZ4_INCLUDE_DIR /nix/store/z08rawrcd8bdfsj1w29w0ab8d01fizm8-lz4-1.10.0-dev/include
-- ICU_VERSION (system) is 74
-- ICU_INCLUDE_DIRS /nix/store/g9fmij5q8yrd6ljdfxl3gslaxms53qlq-icu4c-74.2-dev/include
-- ICU_LIBRARIES /nix/store/x14h84400276pwrylz92aswcgpfl072i-icu4c-74.2/lib/libicuuc.dylib;/nix/store/x14h84400276pwrylz92aswcgpfl072i-icu4c-74.2/lib/libicuio.dylib;/nix/store/x14h84400276pwrylz92aswcgpfl072i-icu4c-74.2/lib/libicudata.dylib;/nix/store/x14h84400276pwrylz92aswcgpfl072i-icu4c-74.2/lib/libicui18n.dylib
-- Found Protobuf: /nix/store/j6lkcznbpb6878mnrq877fzk34zxk4a3-protobuf-21.12/lib/libprotobuf.dylib (found version "3.21.12")
-- PROTOBUF_VERSION (system) is 3.21.12
-- PROTOBUF_INCLUDE_DIR /nix/store/j6lkcznbpb6878mnrq877fzk34zxk4a3-protobuf-21.12/include
-- PROTOBUF_PROTOC_EXECUTABLE /nix/store/j6lkcznbpb6878mnrq877fzk34zxk4a3-protobuf-21.12/bin/protoc
-- Found BISON: /nix/store/wlayh5mbsx32jsn7kl7d6r74vkdizd0n-bison-3.8.2/bin/bison (found version "3.8.2")
-- Found CURL: /nix/store/ifcj5pbackbazl7b91g64m3kbd4v1q01-curl-8.11.0/lib/libcurl.dylib (found version "8.11.0")
-- CURL_INCLUDE_DIR = /nix/store/i4z3xbailna79bxycv6apy5f6phr3q0v-curl-8.11.0-dev/include
-- CURL_VERSION (system) is 8.11
-- RAPIDJSON_INCLUDE_DIR /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/extra/rapidjson/include
-- RAPIDJSON_MAJOR_VERSION is 1
-- RAPIDJSON_MINOR_VERSION is 1
CMake Warning at cmake/fido2.cmake:188 (MESSAGE):
  WITH_FIDO is set to "none".  FIDO based authentication plugins will be
  skipped.
Call Stack (most recent call first):
  CMakeLists.txt:1956 (MYSQL_CHECK_FIDO)


-- Performing Test HAVE_Wunreachable-code-break
-- Performing Test HAVE_Wunreachable-code-break - Success
-- Performing Test HAVE_Wunreachable-code-return
-- Performing Test HAVE_Wunreachable-code-return - Success
-- Performing Test HAVE_Wstring-concatenation
-- Performing Test HAVE_Wstring-concatenation - Success
-- Performing Test HAVE_Wdocumentation
-- Performing Test HAVE_Wdocumentation - Success
-- Performing Test HAVE_Wno-documentation-deprecated-sync
-- Performing Test HAVE_Wno-documentation-deprecated-sync - Success
-- Performing Test HAVE_CXX_W_inconsistent_missing_destructor_override
-- Performing Test HAVE_CXX_W_inconsistent_missing_destructor_override - Success
-- Performing Test HAVE_CXX_W_inconsistent_missing_override
-- Performing Test HAVE_CXX_W_inconsistent_missing_override - Success
-- Performing Test HAVE_CXX_W_shadow_field
-- Performing Test HAVE_CXX_W_shadow_field - Success
-- Performing Test HAVE_CXX_W_string_concatenation
-- Performing Test HAVE_CXX_W_string_concatenation - Success
-- Performing Test HAVE_CXX_W_documentation
-- Performing Test HAVE_CXX_W_documentation - Success
-- Performing Test HAVE_CXX_W_documentation_deprecated_sync
-- Performing Test HAVE_CXX_W_documentation_deprecated_sync - Success
-- mysys depends on ext::zlib
-- mysys depends on ext::zstd
-- clientlib depends on ext::zlib
-- Found Unix DNS SRV APIs
-- Skipping the LDAP client authentication plugin
-- Skipping the KERBEROS client authentication plugin.
-- Skipping the OCI authentication client plugin.
CMake Warning at libmysql/fido_client/common/CMakeLists.txt:26 (MESSAGE):
  Skipping the fido_client_common library.


-- Skipping webauthn client authentication plugin.
-- MERGE_CONVENIENCE_LIBRARIES TARGET mysqlclient
-- MERGE_CONVENIENCE_LIBRARIES LIBS clientlib;mytime;strings;vio;mysys;zlib
-- Looking for sched_getcpu
-- Looking for sched_getcpu - not found
-- Looking for nanosleep
-- Looking for nanosleep - found
-- Performing Test HAVE_FALLOC_PUNCH_HOLE_AND_KEEP_SIZE
-- Performing Test HAVE_FALLOC_PUNCH_HOLE_AND_KEEP_SIZE - Failed
-- Performing Test HAVE_FALLOC_FL_ZERO_RANGE
-- Performing Test HAVE_FALLOC_FL_ZERO_RANGE - Failed
-- Performing Test HAVE_IB_GCC_SYNC_SYNCHRONISE
-- Performing Test HAVE_IB_GCC_SYNC_SYNCHRONISE - Success
-- Performing Test HAVE_IB_GCC_ATOMIC_THREAD_FENCE
-- Performing Test HAVE_IB_GCC_ATOMIC_THREAD_FENCE - Success
-- Performing Test HAVE_IB_LINUX_FUTEX
-- Performing Test HAVE_IB_LINUX_FUTEX - Failed
-- Performing Test HAVE_CXX_W_logical_op
-- Performing Test HAVE_CXX_W_logical_op - Failed
-- Performing Test HAVE_CXX_W_cast_function_type
-- Performing Test HAVE_CXX_W_cast_function_type - Success
-- Performing Test HAVE_CXX_W_unused_but_set_variable
-- Performing Test HAVE_CXX_W_unused_but_set_variable - Success
-- Building ndbcluster plugin
-- Looking for pthread_self
-- Looking for pthread_self - found
-- Looking for sched_get_priority_min
-- Looking for sched_get_priority_min - found
-- Looking for sched_get_priority_max
-- Looking for sched_get_priority_max - found
-- Looking for sched_setaffinity
-- Looking for sched_setaffinity - not found
-- Looking for sched_setscheduler
-- Looking for sched_setscheduler - not found
-- Looking for processor_bind
-- Looking for processor_bind - not found
-- Looking for epoll_create
-- Looking for epoll_create - not found
-- Looking for sysconf
-- Looking for sysconf - found
-- Looking for atomic_swap_32
-- Looking for atomic_swap_32 - not found
-- Looking for pthread_mutexattr_init
-- Looking for pthread_mutexattr_init - found
-- Looking for pthread_mutexattr_settype
-- Looking for pthread_mutexattr_settype - found
-- Looking for pthread_mutexattr_setpshared
-- Looking for pthread_mutexattr_setpshared - found
-- Looking for pthread_setschedparam
-- Looking for pthread_setschedparam - found
-- Looking for priocntl
-- Looking for priocntl - not found
-- Looking for processor_affinity
-- Looking for processor_affinity - not found
-- Looking for cpuset_setaffinity
-- Looking for cpuset_setaffinity - not found
-- Looking for setpriority
-- Looking for setpriority - found
-- Looking for bzero
-- Looking for bzero - found
-- Looking for explicit_bzero
-- Looking for explicit_bzero - not found
-- Looking for memset_s
-- Looking for memset_s - found
-- Looking for ffs
-- Looking for ffs - found
-- Looking for include file atomic.h
-- Looking for include file atomic.h - not found
-- Looking for include file Processtopologyapi.h
-- Looking for include file Processtopologyapi.h - not found
-- Looking for include file Processthreadsapi.h
-- Looking for include file Processthreadsapi.h - not found
-- Looking for include file ncursesw/curses.h
-- Looking for include file ncursesw/curses.h - found
-- Looking for include file ncursesw.h
-- Looking for include file ncursesw.h - not found
-- Looking for include file ncurses.h
-- Looking for include file ncurses.h - found
-- Looking for include file ncurses/curses.h
-- Looking for include file ncurses/curses.h - found
-- Looking for include file xfs/xfs.h
-- Looking for include file xfs/xfs.h - not found
-- Performing Test HAVE___BUILTIN_FFS
-- Performing Test HAVE___BUILTIN_FFS - Success
-- Performing Test HAVE___BUILTIN_CTZ
-- Performing Test HAVE___BUILTIN_CTZ - Success
-- Performing Test HAVE___BUILTIN_CLZ
-- Performing Test HAVE___BUILTIN_CLZ - Success
-- Performing Test HAVE__BITSCANFORWARD
-- Performing Test HAVE__BITSCANFORWARD - Failed
-- Performing Test HAVE__BITSCANREVERSE
-- Performing Test HAVE__BITSCANREVERSE - Failed
-- Performing Test HAVE_MAC_OS_X_THREAD_INFO
-- Performing Test HAVE_MAC_OS_X_THREAD_INFO - Success
-- Performing Test HAVE_LINUX_SCHEDULING
-- Performing Test HAVE_LINUX_SCHEDULING - Failed
-- Performing Test HAVE_SOLARIS_AFFINITY
-- Performing Test HAVE_SOLARIS_AFFINITY - Failed
-- Performing Test HAVE_LINUX_FUTEX
-- Performing Test HAVE_LINUX_FUTEX - Failed
-- Looking for getrlimit
-- Looking for getrlimit - found
-- Excluding Cluster Java components
-- Check size of char
-- Check size of char - done
-- Check size of short
-- Check size of short - done
-- Check size of int
-- Check size of int - done
-- Check size of long
-- Check size of long - done
-- Check size of char*
-- Check size of char* - done
-- Check size of long long
-- Check size of long long - done
-- ndbgeneral depends on ext::zlib
-- Performing Test HAVE_CXX_W_error_maybe_uninitialized
-- Performing Test HAVE_CXX_W_error_maybe_uninitialized - Failed
-- Performing Test HAVE_CXX_W_format_overflow
-- Performing Test HAVE_CXX_W_format_overflow - Success
-- ndbapi depends on ext::zlib
-- LIBS_TO_MERGE mysys;mytime;ndbapi;ndbgeneral;ndblogger;ndbmgmapi;ndbmgmcommon;ndbportlib;ndbsignaldata;ndbtrace;ndbtransport;strings;zlib
-- MERGE_CONVENIENCE_LIBRARIES TARGET ndbclient_static
-- MERGE_CONVENIENCE_LIBRARIES LIBS mysys;mytime;ndbapi;ndbgeneral;ndblogger;ndbmgmapi;ndbmgmcommon;ndbportlib;ndbsignaldata;ndbtrace;ndbtransport;strings;zlib
-- Performing Test HAVE_PEERCRED
-- Performing Test HAVE_PEERCRED - Failed
-- Performing Test HAVE_CXX_W_extra_semi
-- Performing Test HAVE_CXX_W_extra_semi - Success
-- Performing Test HAVE_XDR_OPS_X_PUTINT32
-- Performing Test HAVE_XDR_OPS_X_PUTINT32 - Failed
-- Performing Test HAVE_XDR_OPS_X_GETINT32
-- Performing Test HAVE_XDR_OPS_X_GETINT32 - Failed
-- Performing Test HAVE_RPC_INLINE_T
-- Performing Test HAVE_RPC_INLINE_T - Failed
-- Performing Test HAVE_CXX_W_unused_but_set_parameter
-- Performing Test HAVE_CXX_W_unused_but_set_parameter - Success
-- Performing Test HAVE_CXX_W_suggest_override
-- Performing Test HAVE_CXX_W_suggest_override - Success
-- MYSQLX - Text log of protobuf messages enabled
-- json_client_library depends on ext::zlib
-- MERGE_CONVENIENCE_LIBRARIES TARGET json_binlog_static
-- MERGE_CONVENIENCE_LIBRARIES LIBS json_client_library;mysys;mytime;strings;decimal
-- Performing Test HAVE_CXX_W_unused_variable
-- Performing Test HAVE_CXX_W_unused_variable - Success
-- Performing Test HAVE_NO_BUILTIN_MEMCMP
-- Performing Test HAVE_NO_BUILTIN_MEMCMP - Success
-- BISON outputs /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/build/sql/sql_yacc.cc;/tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/build/sql/sql_yacc.h
-- BISON outputs /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/build/sql/sql_hints.yy.cc;/tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/build/sql/sql_hints.yy.h
-- CONFIG_CLIENT_LIBS -lzstd  -lssl  -lcrypto -lresolv
-- CONFIG_LIBS_PRIVATE -lzstd  -lssl  -lcrypto -lresolv
-- INSTALL mysqlclient.pc lib/pkgconfig
/nix/store/hazx1rlhrn1vng11rma8dv941xm3wh6v-developer_cmds-79/bin/ctags: illegal option -- -
usage: ctags [-BFTaduwvx] [-f tagsfile] file ...
-- MAKE_DIRECTORY /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/build/lib/mysql
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- CMAKE_BUILD_TYPE: Release
-- COMPILE_DEFINITIONS: __STDC_LIMIT_MACROS;__STDC_FORMAT_MACROS;_USE_MATH_DEFINES;LZ4_DISABLE_DEPRECATE_WARNINGS;HAVE_TLSv13
-- CMAKE_C_FLAGS: -fno-omit-frame-pointer -ftls-model=initial-exec  -Wall -Wextra -Wformat-security -Wvla -Wundef -Wmissing-format-attribute -Wwrite-strings -Wconditional-uninitialized -Wextra-semi -Wmissing-noreturn -Wunreachable-code-break -Wunreachable-code-return -Wstring-concatenation -Wdocumentation -Wno-documentation-deprecated-sync
-- CMAKE_CXX_FLAGS: -std=c++20 -fno-omit-frame-pointer -ftls-model=initial-exec  -Wall -Wextra -Wformat-security -Wvla -Wundef -Wmissing-format-attribute -Woverloaded-virtual -Wcast-qual -Wno-null-conversion -Wno-unused-private-field -Wconditional-uninitialized -Wdeprecated -Wno-deprecated-declarations -Wno-shorten-64-to-32 -Wextra-semi -Wheader-hygiene -Wnon-virtual-dtor -Wundefined-reinterpret-cast -Wrange-loop-analysis -Winconsistent-missing-destructor-override -Winconsistent-missing-override -Wshadow-field -Wstring-concatenation -Wdocumentation -Wno-documentation-deprecated-sync
-- CMAKE_CXX_FLAGS_DEBUG: -DSAFE_MUTEX -DENABLED_DEBUG_SYNC -g
-- CMAKE_CXX_FLAGS_RELWITHDEBINFO: -ffunction-sections -fdata-sections -O2 -g -DNDEBUG
-- CMAKE_CXX_FLAGS_RELEASE: -ffunction-sections -fdata-sections -O3 -DNDEBUG
-- CMAKE_CXX_FLAGS_MINSIZEREL: -ffunction-sections -fdata-sections -Os -DNDEBUG
-- CMAKE_C_LINK_FLAGS: -Wl,-search_paths_first -Wl,-headerpad_max_install_names
-- CMAKE_CXX_LINK_FLAGS: -Wl,-search_paths_first -Wl,-headerpad_max_install_names
-- CMAKE_EXE_LINKER_FLAGS 
-- CMAKE_MODULE_LINKER_FLAGS 
-- CMAKE_SHARED_LINKER_FLAGS 
-- Configuring done (133.4s)
-- Generating done (1.5s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/build
cmake: enabled parallel building
cmake: enabled parallel installing
configurePhase completed in 2 minutes 25 seconds
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j10 SHELL=/nix/store/5wjj1f3pa1ndi3kvskm1fcs9kdy2xajf-bash-5.2p37/bin/bash
[  0%] Building CXX object extra/libcno/CMakeFiles/cno_huffman_generator.dir/cno_huffman_generator.cc.o
[  0%] Generating Docs/INFO_SRC
[  0%] Building CXX object vio/CMakeFiles/vio.dir/vio.cc.o
[  0%] Building CXX object libs/mysql/serialization/CMakeFiles/mysql_serialization.dir/archive_binary.cpp.o
[  0%] Generating Docs/INFO_BIN
[  0%] Building CXX object libs/mysql/gtid/CMakeFiles/mysql_gtid.dir/gtid.cpp.o
[  0%] Building C object extra/zlib/zlib-1.3.1/CMakeFiles/zlib.dir/adler32.c.o
[  0%] Building CXX object strings/CMakeFiles/uca9dump.dir/uca9-dump.cc.o
[  0%] Building CXX object mysys/CMakeFiles/mytime.dir/my_time.cc.o
[  0%] Building C object CMakeFiles/stack_direction.dir/cmake/stack_direction.c.o
[  0%] Built target INFO_SRC
[  0%] Built target INFO_BIN
[  0%] Running C++ protobuf compiler on /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/sql/protobuf/replication_asynchronous_connection_failover.proto - lite version
[  0%] Linking C executable runtime_output_directory/stack_direction
[  0%] Building C object extra/zlib/zlib-1.3.1/CMakeFiles/zlib.dir/compress.c.o
[  0%] Building CXX object sql-common/oci/CMakeFiles/oci_common.dir/signing_key.cc.o
[  0%] Building CXX object sql/protobuf/CMakeFiles/rpl_protobuf_lite.dir/generated/protobuf_lite/replication_asynchronous_connection_failover.pb.cc.o
[  0%] Building C object extra/zlib/zlib-1.3.1/CMakeFiles/zlib.dir/crc32.c.o
[  0%] Built target stack_direction
[  0%] Building C object extra/zlib/zlib-1.3.1/CMakeFiles/zlib.dir/deflate.c.o
[  0%] transform_to_lite /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/plugin/x/protocol/protobuf/mysqlx.proto
In file included from /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/libs/mysql/serialization/archive_binary.cpp:24:
In file included from /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/libs/mysql/serialization/archive_binary.h:29:
/tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/libs/mysql/serialization/archive.h:84:43: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
   84 |     return Archive_derived_type::template get_size(std::forward<Type>(arg));
      |                                           ^
/tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/libs/mysql/serialization/archive.h:91:43: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
   91 |     return Archive_derived_type::template get_size_written();
      |                                           ^
[  0%] Building CXX object vio/CMakeFiles/vio.dir/viosocket.cc.o
In file included from /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/libs/mysql/gtid/gtid.cpp:24:
In file included from /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/libs/mysql/gtid/gtid.h:36:
In file included from /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/libs/mysql/serialization/archive_binary.h:29:
/tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/libs/mysql/serialization/archive.h:84:43: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
   84 |     return Archive_derived_type::template get_size(std::forward<Type>(arg));
      |                                           ^
/tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/libs/mysql/serialization/archive.h:91:43: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
   91 |     return Archive_derived_type::template get_size_written();
      |                                           ^
2 errors generated.
make[2]: *** [libs/mysql/serialization/CMakeFiles/mysql_serialization.dir/build.make:76: libs/mysql/serialization/CMakeFiles/mysql_serialization.dir/archive_binary.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3732: libs/mysql/serialization/CMakeFiles/mysql_serialization.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
2 errors generated.
[  0%] transform_to_lite /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/plugin/x/protocol/protobuf/mysqlx_connection.proto
make[2]: *** [libs/mysql/gtid/CMakeFiles/mysql_gtid.dir/build.make:76: libs/mysql/gtid/CMakeFiles/mysql_gtid.dir/gtid.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3758: libs/mysql/gtid/CMakeFiles/mysql_gtid.dir/all] Error 2
[  0%] transform_to_lite /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/plugin/x/protocol/protobuf/mysqlx_crud.proto
[  0%] transform_to_lite /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/plugin/x/protocol/protobuf/mysqlx_cursor.proto
[  0%] transform_to_lite /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/plugin/x/protocol/protobuf/mysqlx_datatypes.proto
[  0%] Building C object extra/zlib/zlib-1.3.1/CMakeFiles/zlib.dir/gzclose.c.o
[  0%] transform_to_lite /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/plugin/x/protocol/protobuf/mysqlx_expect.proto
[  0%] Building C object extra/zlib/zlib-1.3.1/CMakeFiles/zlib.dir/gzlib.c.o
[  0%] transform_to_lite /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/plugin/x/protocol/protobuf/mysqlx_expr.proto
[  0%] transform_to_lite /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/plugin/x/protocol/protobuf/mysqlx_notice.proto
[  0%] transform_to_lite /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/plugin/x/protocol/protobuf/mysqlx_prepare.proto
[  0%] Building C object extra/zlib/zlib-1.3.1/CMakeFiles/zlib.dir/gzread.c.o
[  1%] transform_to_lite /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/plugin/x/protocol/protobuf/mysqlx_resultset.proto
[  1%] Building CXX object sql-common/oci/CMakeFiles/oci_common.dir/ssl.cc.o
[  1%] Building C object extra/zlib/zlib-1.3.1/CMakeFiles/zlib.dir/gzwrite.c.o
[  2%] Building CXX object mysys/CMakeFiles/mytime.dir/my_systime.cc.o
[  2%] transform_to_lite /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/plugin/x/protocol/protobuf/mysqlx_session.proto
[  2%] transform_to_lite /tmp/nix-build-mysql-8.4.3.drv-0/mysql-8.4.3/plugin/x/protocol/protobuf/mysqlx_sql.proto
[  2%] Linking CXX executable ../../runtime_output_directory/cno_huffman_generator
[  2%] Building C object extra/zlib/zlib-1.3.1/CMakeFiles/zlib.dir/inflate.c.o
[  2%] Building C object extra/zlib/zlib-1.3.1/CMakeFiles/zlib.dir/infback.c.o
[  2%] Built target GenLiteProtos
[  2%] Building C object extra/zlib/zlib-1.3.1/CMakeFiles/zlib.dir/inftrees.c.o
[  2%] Linking CXX executable ../runtime_output_directory/uca9dump
[  2%] Building C object extra/zlib/zlib-1.3.1/CMakeFiles/zlib.dir/inffast.c.o
[  2%] Building CXX object vio/CMakeFiles/vio.dir/viossl.cc.o
[  2%] Building C object extra/zlib/zlib-1.3.1/CMakeFiles/zlib.dir/trees.c.o
[  2%] Building C object extra/zlib/zlib-1.3.1/CMakeFiles/zlib.dir/uncompr.c.o
[  2%] Built target cno_huffman_generator
[  2%] Building CXX object vio/CMakeFiles/vio.dir/viosslfactories.cc.o
[  2%] Building C object extra/zlib/zlib-1.3.1/CMakeFiles/zlib.dir/zutil.c.o
[  2%] Building CXX object sql-common/oci/CMakeFiles/oci_common.dir/utilities.cc.o
[  2%] Linking CXX static library librpl_protobuf_lite.a
[  2%] Built target uca9dump
[  3%] Linking C static library ../../../archive_output_directory/libzlib.a
[  3%] Built target rpl_protobuf_lite
[  3%] Built target zlib
[  3%] Linking CXX static library ../archive_output_directory/libmytime.a
[  3%] Built target mytime
[  3%] Linking CXX static library ../archive_output_directory/libvio.a
[  3%] Built target vio
[  3%] Linking CXX static library ../../archive_output_directory/liboci_common.a
[  3%] Built target oci_common
make: *** [Makefile:156: all] Error 2

```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.2.0, macOS 15.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.12.0`
 - channels(rsynnest): `"home-manager"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

Maintainers: @orivej @shyim 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/servers/sql/mysql/8.0.x.nix
+++ b/pkgs/servers/sql/mysql/8.0.x.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchurl,
+  fetchpatch,
   bison,
   cmake,
   pkg-config,
@@ -47,12 +48,20 @@ stdenv.mkDerivation (finalAttrs: {
 
   patches = [
     ./no-force-outline-atomics.patch # Do not force compilers to turn on -moutline-atomics switch
+    # Fix compilation with LLVM 19, adapted from https://github.com/mysql/mysql-server/commit/3a51d7fca76e02257f5c42b6a4fc0c5426bf0421
+    # in https://github.com/NixOS/nixpkgs/pull/374591#issuecomment-2615855076
+    ./libcpp-fixes.patch
+    (fetchpatch {
+      url = "https://github.com/mysql/mysql-server/commit/4a5c00d26f95faa986ffed7a15ee15e868e9dcf2.patch";
+      hash = "sha256-MEl1lQlDYtFjHk0+S02RQFnxMr+YeFxAyNjpDtVHyeE=";
+    })
   ];
 
   ## NOTE: MySQL upstream frequently twiddles the invocations of libtool. When updating, you might proactively grep for libtool references.
-  postPatch = ''
-    substituteInPlace cmake/libutils.cmake --replace /usr/bin/libtool libtool
-    substituteInPlace cmake/os/Darwin.cmake --replace /usr/bin/libtool libtool
+  postPatch = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    substituteInPlace cmake/libutils.cmake --replace-fail /usr/bin/libtool ${cctools}/bin/libtool
+    substituteInPlace cmake/os/Darwin.cmake --replace-fail /usr/bin/libtool ${cctools}/bin/libtool
+    substituteInPlace cmake/package_name.cmake --replace-fail "COMMAND sw_vers" "COMMAND ${DarwinTools}/bin/sw_vers"
   '';
 
   buildInputs =
@@ -77,10 +86,8 @@ stdenv.mkDerivation (finalAttrs: {
       libtirpc
     ]
     ++ lib.optionals stdenv.hostPlatform.isDarwin [
-      cctools
       CoreServices
       developer_cmds
-      DarwinTools
     ];
 
   strictDeps = true;

--- a/pkgs/servers/sql/mysql/libcpp-fixes.patch
+++ b/pkgs/servers/sql/mysql/libcpp-fixes.patch
@@ -1,0 +1,183 @@
+diff --git a/include/my_char_traits.h b/include/my_char_traits.h
+new file mode 100644
+index 00000000..6336bc03
+--- /dev/null
++++ b/include/my_char_traits.h
+@@ -0,0 +1,65 @@
++/* Copyright (c) 2024, Oracle and/or its affiliates.
++
++   This program is free software; you can redistribute it and/or modify
++   it under the terms of the GNU General Public License, version 2.0,
++   as published by the Free Software Foundation.
++
++   This program is designed to work with certain software (including
++   but not limited to OpenSSL) that is licensed under separate terms,
++   as designated in a particular file or component or in included license
++   documentation.  The authors of MySQL hereby grant you an additional
++   permission to link the program and your derivative works with the
++   separately licensed software that they have either included with
++   the program or referenced in the documentation.
++
++   This program is distributed in the hope that it will be useful,
++   but WITHOUT ANY WARRANTY; without even the implied warranty of
++   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
++   GNU General Public License, version 2.0, for more details.
++
++   You should have received a copy of the GNU General Public License
++   along with this program; if not, write to the Free Software
++   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA */
++
++#ifndef MY_CHAR_TRAITS_INCLUDED
++#define MY_CHAR_TRAITS_INCLUDED
++
++#include <cstring>
++
++template <class CharT>
++struct my_char_traits;
++
++/*
++  This is a standards-compliant, drop-in replacement for
++    std::char_traits<unsigned char>
++  We need this because clang libc++ is removing support for it in clang 19.
++  It is not a complete implementation. Rather we implement just enough to
++  compile any usage of char_traits<uchar> we have in our codebase.
++ */
++template <>
++struct my_char_traits<unsigned char> {
++  using char_type = unsigned char;
++  using int_type = unsigned int;
++
++  static void assign(char_type &c1, const char_type &c2) { c1 = c2; }
++
++  static char_type *assign(char_type *s, std::size_t n, char_type a) {
++    return static_cast<char_type *>(memset(s, a, n));
++  }
++
++  static int compare(const char_type *s1, const char_type *s2, std::size_t n) {
++    return memcmp(s1, s2, n);
++  }
++
++  static char_type *move(char_type *s1, const char_type *s2, std::size_t n) {
++    if (n == 0) return s1;
++    return static_cast<char_type *>(memmove(s1, s2, n));
++  }
++
++  static char_type *copy(char_type *s1, const char_type *s2, std::size_t n) {
++    if (n == 0) return s1;
++    return static_cast<char_type *>(memcpy(s1, s2, n));
++  }
++};
++
++#endif  // MY_CHAR_TRAITS_INCLUDED
+diff --git a/sql/mdl_context_backup.h b/sql/mdl_context_backup.h
+index 89e7e23d..cf9c307e 100644
+--- a/sql/mdl_context_backup.h
++++ b/sql/mdl_context_backup.h
+@@ -28,6 +28,7 @@
+ #include <map>
+ #include <memory>
+ 
++#include "my_char_traits.h"
+ #include "sql/malloc_allocator.h"
+ #include "sql/mdl.h"
+ 
+@@ -47,7 +48,8 @@ class MDL_context_backup_manager {
+   /**
+     Key for uniquely identifying MDL_context in the MDL_context_backup map.
+   */
+-  typedef std::basic_string<uchar> MDL_context_backup_key;
++  using MDL_context_backup_key =
++      std::basic_string<uchar, my_char_traits<uchar>>;
+ 
+   class MDL_context_backup;
+ 
+diff --git a/sql/stream_cipher.h b/sql/stream_cipher.h
+index 606d4064..358fbb41 100644
+--- a/sql/stream_cipher.h
++++ b/sql/stream_cipher.h
+@@ -28,6 +28,8 @@
+ #include <memory>
+ #include <string>
+ 
++#include "my_char_traits.h"
++
+ /**
+   @file stream_cipher.h
+ 
+@@ -35,7 +37,8 @@
+          binary log files.
+ */
+ 
+-typedef std::basic_string<unsigned char> Key_string;
++using Key_string =
++    std::basic_string<unsigned char, my_char_traits<unsigned char>>;
+ 
+ /**
+   @class Stream_cipher
+diff --git a/unittest/gunit/binlogevents/transaction_compression-t.cc b/unittest/gunit/binlogevents/transaction_compression-t.cc
+index ba13f979..01af0e3a 100644
+--- a/unittest/gunit/binlogevents/transaction_compression-t.cc
++++ b/unittest/gunit/binlogevents/transaction_compression-t.cc
+@@ -23,6 +23,7 @@
+ */
+ 
+ #include <array>
++#include <string>
+ 
+ #include <gtest/gtest.h>
+ #include "libbinlogevents/include/binary_log.h"
+@@ -51,14 +52,13 @@ class TransactionPayloadCompressionTest : public ::testing::Test {
+   using Managed_buffer_t = Decompressor_t::Managed_buffer_t;
+   using Size_t = Decompressor_t::Size_t;
+   using Char_t = Decompressor_t::Char_t;
+-  using String_t = std::basic_string<Char_t>;
+   using Decompress_status_t =
+       binary_log::transaction::compression::Decompress_status;
+   using Compress_status_t =
+       binary_log::transaction::compression::Compress_status;
+ 
+-  static String_t constant_data(Size_t size) {
+-    return String_t(size, (Char_t)'a');
++  static std::string constant_data(Size_t size) {
++    return std::string(size, (Char_t)'a');
+   }
+ 
+  protected:
+@@ -69,7 +69,7 @@ class TransactionPayloadCompressionTest : public ::testing::Test {
+   void TearDown() override {}
+ 
+   static void compression_idempotency_test(Compressor_t &c, Decompressor_t &d,
+-                                           String_t data) {
++                                           const std::string &data) {
+     auto debug_string = concat(
+         binary_log::transaction::compression::type_to_string(c.get_type_code()),
+         " ", data.size());
+@@ -104,8 +104,8 @@ class TransactionPayloadCompressionTest : public ::testing::Test {
+ 
+     // Check decompressed data
+     ASSERT_EQ(managed_buffer.read_part().size(), data.size()) << debug_string;
+-    ASSERT_EQ(data, String_t(managed_buffer.read_part().begin(),
+-                             managed_buffer.read_part().end()))
++    ASSERT_EQ(data, std::string(managed_buffer.read_part().begin(),
++                                managed_buffer.read_part().end()))
+         << debug_string;
+ 
+     // Check that we reached EOF
+@@ -118,7 +118,7 @@ TEST_F(TransactionPayloadCompressionTest, CompressDecompressZstdTest) {
+   for (auto size : buffer_sizes) {
+     binary_log::transaction::compression::Zstd_dec d;
+     binary_log::transaction::compression::Zstd_comp c;
+-    String_t data{TransactionPayloadCompressionTest::constant_data(size)};
++    std::string data{TransactionPayloadCompressionTest::constant_data(size)};
+     TransactionPayloadCompressionTest::compression_idempotency_test(c, d, data);
+     c.set_compression_level(22);
+     TransactionPayloadCompressionTest::compression_idempotency_test(c, d, data);
+@@ -129,7 +129,7 @@ TEST_F(TransactionPayloadCompressionTest, CompressDecompressNoneTest) {
+   for (auto size : buffer_sizes) {
+     binary_log::transaction::compression::None_dec d;
+     binary_log::transaction::compression::None_comp c;
+-    String_t data{TransactionPayloadCompressionTest::constant_data(size)};
++    std::string data{TransactionPayloadCompressionTest::constant_data(size)};
+     TransactionPayloadCompressionTest::compression_idempotency_test(c, d, data);
+   }
+ }

